### ISSUE TITLE
If lr0 == lrf, should learning rate stay the same? 

### ISSUE THREAD

**Issue by mazatov** [ultralytics/ultralytics#17494]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I'm training Yolo11 model on my dataset with default learning rate parameters. 
```
lr0: 0.01
lrf: 0.01
momentum: 0.937
weight_decay: 0.0005
warmup_epochs: 3.0
warmup_momentum: 0.8
warmup_bias_lr: 0.1
``` 

I'm a bit confused what happens with learning rate when it receives this parameters. One one hand `lr0==lrf` which indicates that initial learning rate is the same as final, so it should stay the same throughout. On the other hand there are all these decay parameters also specified. 

When I look at results.csv, I can see that `lr/pg0,lr/pg1,lr/pg2` parameters seem to be decaying. So am a bit confused what is happening with learning rate. 

``` 
lr/pg0 | lr/pg1 | lr/pg2
-- | -- | --
0.003332 | 0.003332 | 0.003332
0.006652 | 0.006652 | 0.006652
0.009959 | 0.009959 | 0.009959
0.009941 | 0.009941 | 0.009941
0.009921 | 0.009921 | 0.009921
0.009901 | 0.009901 | 0.009901
0.009881 | 0.009881 | 0.009881
0.009861 | 0.009861 | 0.009861
0.009842 | 0.009842 | 0.009842
0.009822 | 0.009822 | 0.009822
0.009802 | 0.009802 | 0.009802
0.009782 | 0.009782 | 0.009782
0.009762 | 0.009762 | 0.009762
0.009743 | 0.009743 | 0.009743
0.009723 | 0.009723 | 0.009723
0.009703 | 0.009703 | 0.009703
0.009683 | 0.009683 | 0.009683
0.009663 | 0.009663 | 0.009663
0.009644 | 0.009644 | 0.009644
0.009624 | 0.009624 | 0.009624
0.009604 | 0.009604 | 0.009604
0.009584 | 0.009584 | 0.009584
0.009564 | 0.009564 | 0.009564
0.009545 | 0.009545 | 0.009545
0.009525 | 0.009525 | 0.009525
0.009505 | 0.009505 | 0.009505
0.009485 | 0.009485 | 0.009485
0.009465 | 0.009465 | 0.009465
0.009446 | 0.009446 | 0.009446
0.009426 | 0.009426 | 0.009426
0.009406 | 0.009406 | 0.009406
0.009386 | 0.009386 | 0.009386
0.009366 | 0.009366 | 0.009366
0.009347 | 0.009347 | 0.009347
0.009327 | 0.009327 | 0.009327
0.009307 | 0.009307 | 0.009307
0.009287 | 0.009287 | 0.009287
0.009267 | 0.009267 | 0.009267
0.009248 | 0.009248 | 0.009248
0.009228 | 0.009228 | 0.009228
0.009208 | 0.009208 | 0.009208
0.009188 | 0.009188 | 0.009188
0.009168 | 0.009168 | 0.009168
0.009149 | 0.009149 | 0.009149
0.009129 | 0.009129 | 0.009129
0.009109 | 0.009109 | 0.009109
0.009089 | 0.009089 | 0.009089
0.009069 | 0.009069 | 0.009069
0.00905 | 0.00905 | 0.00905
0.00903 | 0.00903 | 0.00903
0.00901 | 0.00901 | 0.00901
0.00899 | 0.00899 | 0.00899
0.00897 | 0.00897 | 0.00897
0.008951 | 0.008951 | 0.008951
0.008931 | 0.008931 | 0.008931
0.008911 | 0.008911 | 0.008911
0.008891 | 0.008891 | 0.008891
0.008871 | 0.008871 | 0.008871
0.008852 | 0.008852 | 0.008852
0.008832 | 0.008832 | 0.008832
0.008812 | 0.008812 | 0.008812
0.008792 | 0.008792 | 0.008792
0.008772 | 0.008772 | 0.008772
0.008753 | 0.008753 | 0.008753
0.008733 | 0.008733 | 0.008733
0.008713 | 0.008713 | 0.008713
0.008693 | 0.008693 | 0.008693
0.008673 | 0.008673 | 0.008673
0.008654 | 0.008654 | 0.008654
0.008634 | 0.008634 | 0.008634
0.008614 | 0.008614 | 0.008614
0.008594 | 0.008594 | 0.008594
0.008574 | 0.008574 | 0.008574
0.008555 | 0.008555 | 0.008555
0.008535 | 0.008535 | 0.008535
0.008515 | 0.008515 | 0.008515
0.008495 | 0.008495 | 0.008495
0.008475 | 0.008475 | 0.008475
0.008456 | 0.008456 | 0.008456
0.008436 | 0.008436 | 0.008436
0.008416 | 0.008416 | 0.008416
0.008396 | 0.008396 | 0.008396
0.008376 | 0.008376 | 0.008376
0.008357 | 0.008357 | 0.008357
0.008337 | 0.008337 | 0.008337
0.008317 | 0.008317 | 0.008317
0.008297 | 0.008297 | 0.008297
0.008277 | 0.008277 | 0.008277
0.008258 | 0.008258 | 0.008258
0.008238 | 0.008238 | 0.008238
0.008218 | 0.008218 | 0.008218
0.008198 | 0.008198 | 0.008198
0.008178 | 0.008178 | 0.008178
0.008159 | 0.008159 | 0.008159
0.008139 | 0.008139 | 0.008139
0.008119 | 0.008119 | 0.008119
0.008099 | 0.008099 | 0.008099
0.008079 | 0.008079 | 0.008079
0.00806 | 0.00806 | 0.00806
0.00804 | 0.00804 | 0.00804
0.00802 | 0.00802 | 0.00802
0.008 | 0.008 | 0.008
0.00798 | 0.00798 | 0.00798
0.007961 | 0.007961 | 0.007961
0.007941 | 0.007941 | 0.007941
0.007921 | 0.007921 | 0.007921
0.007901 | 0.007901 | 0.007901
0.007881 | 0.007881 | 0.007881
0.007862 | 0.007862 | 0.007862
0.007842 | 0.007842 | 0.007842
0.007822 | 0.007822 | 0.007822
0.007802 | 0.007802 | 0.007802
0.007782 | 0.007782 | 0.007782
0.007763 | 0.007763 | 0.007763
0.007743 | 0.007743 | 0.007743
0.007723 | 0.007723 | 0.007723
0.007703 | 0.007703 | 0.007703
0.007683 | 0.007683 | 0.007683
0.007664 | 0.007664 | 0.007664
0.007644 | 0.007644 | 0.007644
0.007624 | 0.007624 | 0.007624
0.007604 | 0.007604 | 0.007604
0.007584 | 0.007584 | 0.007584
0.007565 | 0.007565 | 0.007565
0.007545 | 0.007545 | 0.007545
0.007525 | 0.007525 | 0.007525
0.007505 | 0.007505 | 0.007505
0.007485 | 0.007485 | 0.007485
0.007466 | 0.007466 | 0.007466
0.007446 | 0.007446 | 0.007446
0.007426 | 0.007426 | 0.007426
0.007406 | 0.007406 | 0.007406
0.007386 | 0.007386 | 0.007386
0.007367 | 0.007367 | 0.007367
0.007347 | 0.007347 | 0.007347
0.007327 | 0.007327 | 0.007327
0.007307 | 0.007307 | 0.007307
0.007287 | 0.007287 | 0.007287
0.007268 | 0.007268 | 0.007268
0.007248 | 0.007248 | 0.007248
0.007228 | 0.007228 | 0.007228
0.007208 | 0.007208 | 0.007208
0.007188 | 0.007188 | 0.007188
0.007169 | 0.007169 | 0.007169
0.007149 | 0.007149 | 0.007149
0.007129 | 0.007129 | 0.007129
0.007109 | 0.007109 | 0.007109
0.007089 | 0.007089 | 0.007089
0.00707 | 0.00707 | 0.00707
0.00705 | 0.00705 | 0.00705
0.00703 | 0.00703 | 0.00703
0.00701 | 0.00701 | 0.00701
0.00699 | 0.00699 | 0.00699
0.006971 | 0.006971 | 0.006971
0.006951 | 0.006951 | 0.006951
0.006931 | 0.006931 | 0.006931
0.006911 | 0.006911 | 0.006911
0.006891 | 0.006891 | 0.006891
0.006872 | 0.006872 | 0.006872
0.006852 | 0.006852 | 0.006852
0.006832 | 0.006832 | 0.006832
0.006812 | 0.006812 | 0.006812
0.006792 | 0.006792 | 0.006792
0.006773 | 0.006773 | 0.006773
0.006753 | 0.006753 | 0.006753
0.006733 | 0.006733 | 0.006733
0.006713 | 0.006713 | 0.006713
0.006693 | 0.006693 | 0.006693
0.006674 | 0.006674 | 0.006674
0.006654 | 0.006654 | 0.006654
0.006634 | 0.006634 | 0.006634
0.006614 | 0.006614 | 0.006614
0.006594 | 0.006594 | 0.006594
0.006575 | 0.006575 | 0.006575
0.006555 | 0.006555 | 0.006555
0.006535 | 0.006535 | 0.006535
0.006515 | 0.006515 | 0.006515
0.006495 | 0.006495 | 0.006495
0.006476 | 0.006476 | 0.006476
0.006456 | 0.006456 | 0.006456
0.006436 | 0.006436 | 0.006436
0.006416 | 0.006416 | 0.006416
0.006396 | 0.006396 | 0.006396
0.006377 | 0.006377 | 0.006377
0.006357 | 0.006357 | 0.006357
0.006337 | 0.006337 | 0.006337
0.006317 | 0.006317 | 0.006317
0.006297 | 0.006297 | 0.006297
0.006278 | 0.006278 | 0.006278
```

Thanks

**Comment by UltralyticsAssistant** [comment]
👋 Hello @mazatov, thank you for reaching out and sharing your query with Ultralytics 🚀! We recommend taking a look at our comprehensive [Docs](https://docs.ultralytics.com) for a detailed understanding, especially if you're navigating learning rate dynamics in training scenarios. You'll find insightful [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples that might help provide clarity.

If your post is a 🐛 Bug Report, kindly provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us diagnose the issue effectively.

If you're facing a custom training ❓ Question, providing additional details will be very helpful, such as dataset samples or training logs, and ensure you're following our [Model Training Tips](https://docs.ultralytics.com/guides/model-training-tips/) for best practices.

Why not engage with the Ultralytics community? Share insights and get advice on our [Discord](https://ultralytics.com/discord) 🎧 for live chats, or join the debates on [Discourse](https://community.ultralytics.com) and our [Subreddit](https://reddit.com/r/ultralytics).

## Upgrade

Make sure you're using the latest `ultralytics` package along with all necessary [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml). Install or upgrade in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

The YOLO framework can be run in several verified environments, each equipped with necessary dependencies like [CUDA](https://developer.nvidia.com/cuda), [CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. Refer to the [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. Explore the [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

A green badge signifies all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are passing. These Continuous Integration tests ensure the reliable performance of YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) across macOS, Windows, and Ubuntu every 24 hours and subsequent to every commit.

Please note this is an automated response, but rest assured, an Ultralytics engineer will look at your issue shortly and provide further assistance.

**Comment by Y-T-G** [comment]
> ### Search before asking
> 
> - [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.
> 

Duplicate of https://github.com/ultralytics/ultralytics/issues/9104

**Comment by Y-T-G** [comment]
Duplicate of #9104

> ### Search before asking
> 
> - [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.
>
